### DISCUSSION TITLE
Any recommended way to create a dynamic proxy?

### DISCUSSION THREAD

**Discussion by h0cheung** [apache/skywalking#12337] [Q&A]
I'm trying to develop a plugin that needs to intercept a fixed name, type generic field within a known class.

As tested by me, some codes like this works:

```
// onConstruct
Object proxiedObject = new ByteBuddy()
        .subclass(innerObject.getClass())
        .method(ElementMatchers.any())
        .intercept(InvocationHandlerAdapter.of((proxy, method, args) -> {
            // do something before method
            return method.invoke(innerObject, args);
            // do something after method
        }))
        .make()
        .load(innerObject.getClass().getClassLoader())
        .getLoaded().newInstance();
ReflectionUtils.setValue(outerClass.class, objInst, "fieldName", proxiedObject);
```

But this directly calls the bytebuddy api, which is never called by any other plugins. Is this a good way? If not, is there a better way to do that?

**Comment by wu-sheng** [comment]
No, we don't provide that mechanism. In all our supported cases, we don't find out the necessity.
